### ISSUE TITLE
AutoTool crashing with NullPointerException

### ISSUE THREAD

**Issue by LordQuadrato** [ImpactDevelopment/ImpactClient#653]
### What happens?
Impact crashes when trying to use `AutoTool` without the appropriate tool in your inventory.
**This doesn't happen if you already used `AutoTool` successfully.**

### Exact steps to reproduce?

- Don't have a tool in your inventory
- Enable `AutoTool`
- Destroy a block

:x: The game crashes

<details> 
  <summary><strong>Game output</strong></summary>

```
Reported exception thrown!
f: Ticking entity
	at amu.k(SourceFile:1389)
	at bib.t(SourceFile:1691)
	at bib.az(SourceFile:1000)
	at bib.a(SourceFile:419)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
Caused by: java.lang.NullPointerException
	at me.zero.clarinet.d5.lambda$newSlotSwapper$0(d5.java)
	at me.zero.clarinet.d5$$Lambda$654/1486570589.accept(Unknown Source)
	at clientapi.util.interfaces.impl.MergedMutable.accept(MergedMutable.java:49)
	at clientapi.util.interfaces.impl.MutableSwapper.reset(MutableSwapper.java:68)
	at me.zero.clarinet.dj$$Lambda$789/457491266.run(Unknown Source)
	at me.zero.clarinet.dL.t(dL.java)
	at me.zero.clarinet.dj.lambda$new$2(dj.java)
	at me.zero.clarinet.dj$$Lambda$793/1541500074.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.EventManager.lambda$post$5(EventManager.java:91)
	at me.zero.alpine.EventManager$$Lambda$106/453966584.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.EventManager.post(EventManager.java:91)
	at clientapi.event.ClientAPIEventManager.post(ClientAPIEventManager.java:44)
	at bud.handler$onPostUpdate$zzd000(SourceFile:1220)
	at bud.B_(SourceFile:208)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)
	at amu.k(SourceFile:1382)
	... 10 more
```

</details>

<details> 
  <summary><strong>Crash report</strong></summary>
 
```
---- Minecraft Crash Report ----
// Everything's going to plan. No, really, that was supposed to happen.

Time: 7/17/18 10:38 PM
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at me.zero.clarinet.d5.lambda$newSlotSwapper$0(d5.java)
	at me.zero.clarinet.d5$$Lambda$654/126789224.accept(Unknown Source)
	at clientapi.util.interfaces.impl.MergedMutable.accept(MergedMutable.java:49)
	at clientapi.util.interfaces.impl.MutableSwapper.reset(MutableSwapper.java:68)
	at me.zero.clarinet.dj$$Lambda$789/930124685.run(Unknown Source)
	at me.zero.clarinet.dL.t(dL.java)
	at me.zero.clarinet.dj.lambda$new$2(dj.java)
	at me.zero.clarinet.dj$$Lambda$793/1947846882.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.EventManager.lambda$post$5(EventManager.java:91)
	at me.zero.alpine.EventManager$$Lambda$106/2011009230.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.EventManager.post(EventManager.java:91)
	at clientapi.event.ClientAPIEventManager.post(ClientAPIEventManager.java:44)
	at bud.handler$onPostUpdate$zzd000(SourceFile:1220)
	at bud.B_(SourceFile:208)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)
	at amu.k(SourceFile:1382)
	at bib.t(SourceFile:1691)
	at bib.az(SourceFile:1000)
	at bib.a(SourceFile:419)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at me.zero.clarinet.d5.lambda$newSlotSwapper$0(d5.java)
	at me.zero.clarinet.d5$$Lambda$654/126789224.accept(Unknown Source)
	at clientapi.util.interfaces.impl.MergedMutable.accept(MergedMutable.java:49)
	at clientapi.util.interfaces.impl.MutableSwapper.reset(MutableSwapper.java:68)
	at me.zero.clarinet.dj$$Lambda$789/930124685.run(Unknown Source)
	at me.zero.clarinet.dL.t(dL.java)
	at me.zero.clarinet.dj.lambda$new$2(dj.java)
	at me.zero.clarinet.dj$$Lambda$793/1947846882.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.EventManager.lambda$post$5(EventManager.java:91)
	at me.zero.alpine.EventManager$$Lambda$106/2011009230.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.EventManager.post(EventManager.java:91)
	at clientapi.event.ClientAPIEventManager.post(ClientAPIEventManager.java:44)
	at bud.handler$onPostUpdate$zzd000(SourceFile:1220)
	at bud.B_(SourceFile:208)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)

-- Entity being ticked --
Details:
	Entity Type: null (bud)
	Entity ID: 309
	Entity Name: Lord_Quadrato
	Entity's Exact location: -1011.36, 238.00, 142.53
	Entity's Block location: World: (-1012,238,142), Chunk: (at 12,14,14 in -64,8; contains blocks -1024,0,128 to -1009,255,143), Region: (-2,0; contains chunks -64,0 to -33,31, blocks -1024,0,0 to -513,255,511)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at amu.k(SourceFile:1382)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [bud['Lord_Quadrato'/309, l='MpServer', x=-1011.36, y=238.00, z=142.53]]
	Chunk stats: MultiplayerChunkCache: 625, 625
	Level seed: 0
	Level generator: ID 01 - flat, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-1016,4,131), Chunk: (at 8,0,3 in -64,8; contains blocks -1024,0,128 to -1009,255,143), Region: (-2,0; contains chunks -64,0 to -33,31, blocks -1024,0,0 to -513,255,511)
	Level time: 274032 game time, 57227 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 5 total; [zx['Cow'/299, l='MpServer', x=-1083.45, y=237.00, z=128.79], adu['Zombie Villager'/460, l='MpServer', x=-986.37, y=256.00, z=114.98], zx['Cow'/305, l='MpServer', x=-1045.13, y=237.00, z=210.87], aas['Llama'/306, l='MpServer', x=-1036.09, y=237.00, z=208.72], bud['Lord_Quadrato'/309, l='MpServer', x=-1011.36, y=238.00, z=142.53]]
	Retry entities: 0 total; []
	Server brand: vanilla
	Server type: Integrated singleplayer server
Stacktrace:
	at bsb.a(WorldClient.java:532)
	at bib.b(SourceFile:2587)
	at bib.a(SourceFile:428)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 436540528 bytes (416 MB) / 1073741824 bytes (1024 MB) up to 1073741824 bytes (1024 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=16M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.12.2-Impact_4.3_Optifine_E1
	LWJGL: 2.9.4
	OpenGL: Intel(R) HD Graphics 620 GL version 4.4.0 - Build 21.20.16.4534, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
	OptiFine Version: OptiFine_1.12.2_HD_U_E1
	OptiFine Build: 20180628-111228
	Render Distance Chunks: 12
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.4.0 - Build 21.20.16.4534
	OpenGlRenderer: Intel(R) HD Graphics 620
	OpenGlVendor: Intel
	CpuCount: 4
```

</details>

<details> 
  <summary><strong>Additional Client Info</strong></summary>
 
- Impact Version: 4.3 for 1.12.2
- Using Optifine? Yes
- Enabled mods: **`Autotool`**
- Operating System: Windows 10 64-bit

</details>

**Comment by CrypticCalamari** [comment]
Seems to happen with punching 2 high grass blocks and fire. Perhaps those blocks are not accounted for with respect to which tools to use for them so it defaults to null instead of fist?

<details>
<summary>Crash Report</summary>
<p>

```
---- Minecraft Crash Report ----
// I let you down. Sorry :(

Time: 7/24/18 9:48 PM
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at me.zero.clarinet.d5.lambda$newSlotSwapper$0(d5.java)
	at clientapi.util.interfaces.impl.MergedMutable.accept(MergedMutable.java:49)
	at clientapi.util.interfaces.impl.MutableSwapper.reset(MutableSwapper.java:68)
	at me.zero.clarinet.dL.t(dL.java)
	at me.zero.clarinet.dj.lambda$new$2(dj.java)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.EventManager.lambda$post$5(EventManager.java:91)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.EventManager.post(EventManager.java:91)
	at clientapi.event.ClientAPIEventManager.post(ClientAPIEventManager.java:44)
	at bud.handler$onPostUpdate$zzd000(SourceFile:1220)
	at bud.B_(SourceFile:208)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)
	at amu.k(SourceFile:1382)
	at bib.t(SourceFile:1691)
	at bib.az(SourceFile:1000)
	at bib.a(SourceFile:419)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at me.zero.clarinet.d5.lambda$newSlotSwapper$0(d5.java)
	at clientapi.util.interfaces.impl.MergedMutable.accept(MergedMutable.java:49)
	at clientapi.util.interfaces.impl.MutableSwapper.reset(MutableSwapper.java:68)
	at me.zero.clarinet.dL.t(dL.java)
	at me.zero.clarinet.dj.lambda$new$2(dj.java)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.EventManager.lambda$post$5(EventManager.java:91)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.EventManager.post(EventManager.java:91)
	at clientapi.event.ClientAPIEventManager.post(ClientAPIEventManager.java:44)
	at bud.handler$onPostUpdate$zzd000(SourceFile:1220)
	at bud.B_(SourceFile:208)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)

-- Entity being ticked --
Details:
	Entity Type: null (bud)
	Entity ID: 258437
	Entity Name: Gatesunder
	Entity's Exact location: 11.81, 53.00, 27.80
	Entity's Block location: World: (11,53,27), Chunk: (at 11,3,11 in 0,1; contains blocks 0,0,16 to 15,255,31), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at amu.k(SourceFile:1382)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [bud['Gatesunder'/258437, l='MpServer', x=11.81, y=53.00, z=27.80]]
	Chunk stats: MultiplayerChunkCache: 169, 169
	Level seed: 0
	Level generator: ID 02 - largeBiomes, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: World: (0,67,0), Chunk: (at 0,4,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 161036207 game time, 149213316 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 31 total; [bud['Gatesunder'/258437, l='MpServer', x=11.81, y=53.00, z=27.80], adf['Zombie Pigman'/258560, l='MpServer', x=26.50, y=35.00, z=-32.50], adf['Zombie Pigman'/258443, l='MpServer', x=-37.50, y=47.00, z=53.50], adf['Zombie Pigman'/258570, l='MpServer', x=42.50, y=14.00, z=55.50], adf['Zombie Pigman'/258569, l='MpServer', x=12.50, y=10.00, z=-23.50], adf['Zombie Pigman'/258568, l='MpServer', x=10.50, y=10.00, z=-26.50], adf['Zombie Pigman'/258450, l='MpServer', x=-38.19, y=44.00, z=86.72], adf['Zombie Pigman'/258449, l='MpServer', x=-30.22, y=83.00, z=-31.50], adf['Zombie Pigman'/258448, l='MpServer', x=-15.73, y=78.00, z=-17.79], adf['Zombie Pigman'/258453, l='MpServer', x=-5.70, y=61.00, z=106.20], adf['Zombie Pigman'/258462, l='MpServer', x=3.81, y=61.00, z=103.57], adf['Zombie Pigman'/258460, l='MpServer', x=33.93, y=63.88, z=40.03], adf['Zombie Pigman'/258466, l='MpServer', x=29.24, y=80.00, z=-48.51], adf['Zombie Pigman'/258469, l='MpServer', x=25.50, y=10.00, z=17.50], adf['Zombie Pigman'/258468, l='MpServer', x=20.50, y=33.00, z=-24.50], adf['Zombie Pigman'/258474, l='MpServer', x=20.50, y=11.00, z=56.50], adf['Zombie Pigman'/258473, l='MpServer', x=27.50, y=10.00, z=20.50], adf['Zombie Pigman'/258472, l='MpServer', x=26.49, y=10.00, z=20.92], adf['Zombie Pigman'/258478, l='MpServer', x=18.50, y=53.00, z=63.50], adf['Zombie Pigman'/258482, l='MpServer', x=43.50, y=9.00, z=3.50], adf['Zombie Pigman'/258480, l='MpServer', x=18.50, y=8.00, z=70.50], adf['Zombie Pigman'/258487, l='MpServer', x=51.49, y=12.00, z=31.79], adf['Zombie Pigman'/258486, l='MpServer', x=63.50, y=99.00, z=-28.73], adf['Zombie Pigman'/258499, l='MpServer', x=88.50, y=13.00, z=-30.50], adf['Zombie Pigman'/258497, l='MpServer', x=83.33, y=13.00, z=-31.45], adf['Zombie Pigman'/258496, l='MpServer', x=86.50, y=13.00, z=-29.50], adf['Zombie Pigman'/258505, l='MpServer', x=21.50, y=6.00, z=54.50], adf['Zombie Pigman'/258550, l='MpServer', x=26.50, y=13.00, z=54.50], adf['Zombie Pigman'/258559, l='MpServer', x=27.50, y=35.00, z=-31.50], adf['Zombie Pigman'/258558, l='MpServer', x=25.50, y=35.00, z=-32.79], adf['Zombie Pigman'/258557, l='MpServer', x=26.50, y=35.00, z=-31.50]]
	Retry entities: 0 total; []
	Server brand: vanilla
	Server type: Non-integrated multiplayer server
Stacktrace:
	at bsb.a(WorldClient.java:532)
	at bib.b(SourceFile:2587)
	at bib.a(SourceFile:428)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Linux (i386) version 3.16.0-6-686-pae
	Java Version: 1.8.0_111, Oracle Corporation
	Java VM Version: OpenJDK Server VM (mixed mode), Oracle Corporation
	Memory: 144993392 bytes (138 MB) / 404807680 bytes (386 MB) up to 523501568 bytes (499 MB)
	JVM Flags: 5 total; -Xmx512M -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.12.2-Impact_4.3-Optifine
	LWJGL: 2.9.4
	OpenGL: Mesa DRI Intel(R) 945GM x86/MMX/SSE2 GL version 2.1 Mesa 10.3.2, Intel Open Source Technology Center
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because ARB_framebuffer_object is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 2x Intel(R) Core(TM)2 CPU T5500 @ 1.66GHz
	OptiFine Version: OptiFine_1.12.2_HD_U_D3
	OptiFine Build: 20180515-125031
	Render Distance Chunks: 2
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 2.1 Mesa 10.3.2
	OpenGlRenderer: Mesa DRI Intel(R) 945GM x86/MMX/SSE2
	OpenGlVendor: Intel Open Source Technology Center
	CpuCount: 2
````
</p>
</details>

**Comment by LeafHacker** [comment]
Sounds promising

**Comment by ZeroMemes** [comment]
Both crash reports were caused by the same issue. The targeted block does *not* seem to affect the outcome.

Fixed in 4.4.
